src/Type/Php/CtypeDigitFunctionTypeSpecifyingExtension.php

```diff
@@ -0,0 +1,56 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Analyser\SpecifiedTypes;
+use PHPStan\Analyser\TypeSpecifier;
+use PHPStan\Analyser\TypeSpecifierAwareExtension;
+use PHPStan\Analyser\TypeSpecifierContext;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\Accessory\AccessoryNumericStringType;
+use PHPStan\Type\FunctionTypeSpecifyingExtension;
+use PHPStan\Type\IntegerRangeType;
+use PHPStan\Type\UnionType;
+use function strtolower;
+
+class CtypeDigitFunctionTypeSpecifyingExtension implements FunctionTypeSpecifyingExtension, TypeSpecifierAwareExtension
+{
+
+	private TypeSpecifier $typeSpecifier;
+
+	public function isFunctionSupported(FunctionReflection $functionReflection, FuncCall $node, TypeSpecifierContext $context): bool
+	{
+		return strtolower($functionReflection->getName()) === 'ctype_digit'
+			&& !$context->null();
+	}
+
+	public function specifyTypes(FunctionReflection $functionReflection, FuncCall $node, Scope $scope, TypeSpecifierContext $context): SpecifiedTypes
+	{
+		if (!isset($node->getArgs()[0])) {
+			return new SpecifiedTypes();
+		}
+		if ($context->null()) {
+			throw new ShouldNotHappenException();
+		}
+
+		$types = [
+			IntegerRangeType::fromInterval(48, 57), // ASCII-codes for 0-9
+			IntegerRangeType::createAllGreaterThanOrEqualTo(256), // Starting from 256 ints are interpreted as strings
+		];
+
+		if ($context->true()) {
+			$types[] = new AccessoryNumericStringType();
```

Comment:
numeric strings always have to be in combination with a basic type. So this should be IntersectionType of StringType and AccessoryNumericStringType.

Reply:
Done